### ISSUE TITLE
`$this` should be allowed in closures

### ISSUE THREAD

**Issue by sirbrillig** [sirbrillig/phpcs-variable-analysis#19]
The use of `$this` inside a closure should be fine, due to [automatic binding](http://php.net/manual/en/functions.anonymous.php#example-167). I'm not sure why there is an example explicitly marking this as a warning.

Here's a simple example that should report no warnings.

```php
class classWithClosure() {
  public function test_get_source_information_from_theme() {
    doSomething( function() {
      return $this->thing;
    } );
  }
}
```
